recipe/magento2.php

```diff
@@ -219,3 +221,137 @@
 ]);
 
 after('deploy:failed', 'magento:maintenance:disable');
+
+// artifact deployment section
+// settings section
+set('artifact_file', 'artifact.tar.gz');
+set('artifact_dir', 'artifacts');
+set('artifact_excludes_file', 'artifacts/excludes');
+
+set('artifact_path', function () {
+    if (!test('[ -d {{artifact_dir}} ]')) {
+        run('mkdir {{artifact_dir}}');
+    }
+    return get('artifact_dir') . '/' . get('artifact_file');
+});
+
+set('bin/tar', function () {
+    if (commandExist('gtar')) {
+        return which('gtar');
+    } else {
+        return which('tar');
+    }
+});
+
+set('cacheToolPath', function() {
+    return get('cacheTool', '{{current_path}}/bin/cachetool');
+});
+
+// tasks section
+desc('Packages all relevant files in an artifact.');
+task('artifact:package', function() {
+    if (!test('[ -f {{artifact_excludes_file}} ]')) {
+        throw new GracefulShutdownException(
+            "No artifact excludes file provided, provide one at artivacts/excludes or change location"
+        );
+    }
+    run('{{bin/tar}} --exclude-from={{artifact_excludes_file}} -czf {{artifact_path}} .');
+});
+
+desc('Uploads artifact in release folder for extraction.');
+task('artifact:upload', function () {
+    upload(get('artifact_path'), '{{release_path}}');
+});
+
+desc('Extracts artifact in release path.');
+task('artifact:extract', function () {
+    run('{{bin/tar}} -xzpf {{release_path}}/{{artifact_file}} -C {{release_path}}');
+    run('rm -rf {{release_path}}/{{artifact_file}}');
+});
+
+desc('Provides env.php for build.');
```

Comment:
Why do you create an env.php file for the build? There is no database etc, everything we need should be in config.php. At least this should be optional

Reply:
Can second this

Reply:
Ok, the idea behind this is/was the way static assets are generated.

Since on production and dev they differ, I thougth it would be a good Idea, to have them in an `env.php` which broke my deployment. But probably you are right, If `env.php` settings override `conf.php` settings, then we should just omit this part, do they.

Reply:
So, OK, I'll put this out soon then...